packages/engine/src/utils/engineEnvBuilder.ts

```
@@ -639,18 +654,15 @@ function configureWallet(coreConfig: IEngineConfig, envVars: IEngineEnvironmentV
 			type: WalletConnectorType.EntityStorage
 		});
 	} else if (envVars.walletConnector === WalletConnectorType.Iota) {
-		coreConfig.types.walletConnector.push({
-			type: WalletConnectorType.Iota,
-			options: {
-				config: {
-					clientOptions: {
-						url: envVars.iotaNodeEndpoint ?? ""
-					},
-					network: envVars.iotaNetwork ?? "",
-					coinType: Coerce.number(envVars.iotaCoinType)
+		const iotaConfig = getIotaConfig(coreConfig);
+		if (iotaConfig) {
```

This `if` should not be here, we need the configuration to always be added, otherwise the Guards within the constructors will never be triggered and it will just fail silently.